### PR TITLE
Set IsDialog property and AccessibleName property for StatusDialog

### PR DESCRIPTION
**Pull request by LeafShi1**
Related #11456

## Proposed changes

- Add class `StatusDialogAccessibility.cs` and set `IsDialog` property to true and set `AccessibleName` to StatusDialog.Text

## Customer Impact

- The `IsDialog` property and the AccessibleName` property can show correct value

## Regression? 

- No

## Risk

- Minimal

### Before
image

### After
AfterChanges

## Test methodology 

- Manually

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/12046)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Printing/StatusDialogAccessibility.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        public class StatusDialogAccessibility : ControlAccessibleObject
+        {
+            public StatusDialogAccessibility(StatusDialog owner) : base(owner)
+            {
+            }
+
+            internal override VARIANT GetPropertyValue(UIA_PROPERTY_ID propertyID) =>
+                propertyID switch
+                {
+                    UIA_PROPERTY_ID.UIA_IsDialogPropertyId => (VARIANT)true,
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Printing/StatusDialogAccessibility.cs:22]: We have a static `VARIANT.True` that you can use here.

File: src/System.Windows.Forms/src/System/Windows/Forms/Printing/StatusDialogAccessibility.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                    _ => base.GetPropertyValue(propertyID)
+                };
+
+            public override string? Name
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/src/System/Windows/Forms/Printing/StatusDialogAccessibility.cs:26]: I thought that Name was working in NVDA and JAWS, why are you adding it?

**Review by Tanya-Solyanik — COMMENTED** [review]
Please add unit tests for these 2 properties.

File: src/System.Windows.Forms/src/System/Windows/Forms/Printing/StatusDialogAccessibility.cs (hunk 2 above)
Comment (LeafShi1) [reply src/System.Windows.Forms/src/System/Windows/Forms/Printing/StatusDialogAccessibility.cs:26]: > I thought that Name was working in NVDA and JAWS, why are you adding it?

Yes, you are right. The Name property does not need to be rewritten. It was just that the Inspect was stuck during verification yesterday, causing the displayed AssibleName to be wrong, It was displayed as Debug. I verified it again through AI and Inspect. It turned out that my worry was unnecessary. I will delete it.
image

**Comment by LeafShi1** [comment]
> Please add unit tests for these 2 properties.

The [StatusDialog ](https://github.com/dotnet/winforms/blob/dde2dcbe865c6676585f52e4fe0973017cb6d78d/src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintControllerWithStatusDialog.StatusDialog.cs#L12) class is private, so I cannot get it. If we have to test it, I need to change it to internal

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Comment by lonitra** [comment]
> The [StatusDialog ](https://github.com/dotnet/winforms/blob/dde2dcbe865c6676585f52e4fe0973017cb6d78d/src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintControllerWithStatusDialog.StatusDialog.cs#L12) class is private, so I cannot get it. If we have to test it, I need to change it to internal

If we are still wanting to add some unit tests, we could potentially use TestAccessor to access the private class if there is a way to get to it from `PrintControllerWithStatusDialog`.

**Comment by LeafShi1** [comment]
> If we are still wanting to add some unit tests, we could potentially use TestAccessor to access the private class if there is a way to get to it from `PrintControllerWithStatusDialog`.

I tried using TestAccessor but I can't get private classes
image

Using TestAccessor I can access private methods/properties/private variables of public classes but I can't access nested private classes of public classes.

**Comment by MelonWang1** [comment]
@LeafShi1 Verified the issue with .NET SDK 9.0.100-rc.2.24460.10 + private dlls built from Winforms main branch, it is fixed that Set IsDialog property is true. The two tools are a little different for AccessibleName. in Inspect tool, called LegacyAccessible.Name, in  Accessiblity Insights tool, called LegacyAccessiblePattern.Name, both that property is Generating Previews. Is this expected?
Accessiblity Insights:
ai
Inspect:
inspect

**Comment by LeafShi1** [comment]
> @LeafShi1 Verified the issue with .NET SDK 9.0.100-rc.2.24460.10 + private dlls built from Winforms main branch, it is fixed that Set IsDialog property is true. The two tools are a little different for AccessibleName. in Inspect tool, called LegacyAccessible.Name, in Accessiblity Insights tool, called LegacyAccessiblePattern.Name, both that property is Generating Previews. Is this expected? 

Yes, I think it is expected

**Pull request merged**